appclient/src/main/java/org/jboss/as/appclient/service/LazyConnectionContextSelector.java

```diff
@@ -99,14 +101,19 @@ private synchronized void createConnection() {
 
     @Override
     public EJBClientContext getCurrent() {
-        if (clientContext == null) {
-            synchronized (this) {
+        return WildFlySecurityManager.doUnchecked(new PrivilegedAction<EJBClientContext>() {
```

Comment:
This should really be surrounded by a if(WildflySecurity.isChecking()) check, as doUnchecked is relatively slow (and it also avoids an object allocation just invoking directly), so we only want to use it if we have to.

I'm not 100% sure why doUnchecked does not perform a similar check internally (it only checks the checking flag, not if the security manager is null). I will ask @dmlloyd